lib/rules/at-rule-no-deprecated/index.mjs

```diff
@@ -0,0 +1,82 @@
+import postcss from 'postcss';
+
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import { deprecatedAtKeywords } from '../../reference/atKeywords.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'at-rule-no-deprecated';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule) => `Unexpected deprecated at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-no-deprecated',
+	fixable: true,
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAtRules: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) return;
+
+		root.walkAtRules((atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			const { name } = atRule;
+
+			if (optionsMatches(secondaryOptions, 'ignoreAtRules', name)) return;
+
+			if (!deprecatedAtKeywords.has(name.toLowerCase())) return;
+
+			const atName = `@${name}`;
+
+			const atNestFixer = () => {
+				const styleRule = postcss.rule({
+					selector: atRule.params,
+					source: atRule.source,
+				});
+
+				styleRule.append(atRule.nodes);
+
+				atRule.replaceWith(styleRule);
+			};
+
+			const fix = name === 'nest' ? atNestFixer : undefined;
```

Comment:
[question] Is the normalization of the name necessary here? E.g.

```js
const fix = name.toLowerCase() === 'nest' ? atNestFixer : undefined;
```

If needed, avoiding repetition of the `toLowerCase()` calls might be better. E.g.

```js
const normalizedName = name.toLowerCase();

if (!deprecatedAtKeywords.has(normalizedName)) return;

const fix = normalizedName === 'nest' ? atNestFixer : undefined;
```

Reply:
Good catch. It was necessary. Fixed in https://github.com/stylelint/stylelint/pull/8251/commits/54b11d3fa9ef7e9537b4267fb99cecf07a563788.